packages/babel-register/README.md

```diff
@@ -20,7 +20,7 @@ require("babel-register");
 ```
 
 All subsequent files required by node with the extensions `.es6`, `.es`, `.jsx`,
-`.mjs`, and `.js` will be transformed by Babel.
+`.mjs`, `.ts`, `.tsx`, and `.js` will be transformed by Babel.
```

Comment:
Need to be careful with that; `.d.ts` files are metadata _only_; they're not code and cannot become code. Having code in them is a syntax error.

Even if there was some syntax you wanted to transform, having to transform it makes the .d.ts useless as that means TypeScript can't use its metadata as-is.

Reply:
The PR explicitly ignore them, and I just wanted to make it clear. While it doesn't really affect the user, it's a big implementation detail on our side.

Reply:
@loganfsmyth just made a suggestion in slack, maybe could make a new issue for it.

> Logan Smyth [8:22 PM]
I've been thinking about https://github.com/babel/babel/pull/6027 and https://github.com/babel/babel/issues/3741 more generally. It seems like it'd be pretty easy to have babel-core include an `extensions` object in its options, so for example the typescript preset could just say:
```
extensions: {
  ".ts": true,
  ".tsx": true,
  ".d.ts": false,
},
```
in its body to tell Babel to start processing `.ts` files.

We could for instance make `@babel/core` essentially `ignore` a file if its extension isn't given in the config or Babel's default extension list.

My main question at that stage is, does it make sense to keep these as booleans, or to just go all the way and make the extensions be `string | null` with the string value being a `sourceType`? Then we could easily give `.ts` files a `sourceType` of its own sourceType which I've mentioned before as potentially being nice anyway, since we'll probably want that so that the CLI for instance could properly output files with `.ts` extensions if the user enabled the typescript syntax plugin but not the transform.

I know Webpack has more complex logic around file types, so maybe we could consider an approach that that over simple sourceTypes too, hard to say.

Reply:
Yeah, I remember about that. Just an idea @loganfsmyth, what if we stored this configuration in the presets, like the React preset would enable the JSX syntax or TS the TS syntax.